### DISCUSSION TITLE
왜 외부링크를 사용하면 안되요? [치자피즈]

### DISCUSSION THREAD

**Discussion by x4cf** [naverz/zepeto-studio-kor#253] [4. General]
제페토 유저 이자 콘텐츠를 제작중인 **개인**으로써 정확한 이유는 알 수 없습니다.
하지만 외부링크의 위험성과 인터넷 공급업체가 아닌 플랫폼 제작자인 제페토를 생각하면 추측은 해볼 수 있습니다.

**1.무단 업데이트**
 서버와 통신을 한다는건 이미 심사가 끝난 콘텐츠 내의 정보를 수정할 수 있다는 이야기이기도 합니다. 월드 내에서 재생되는 영상이나 이미지를 개인의 서버에서 공급할때, 서버는 언제든 송출되는 영상이나 이미지를 수정할 수 있습니다. 
악의적인 의도를 가진 서버는 영상이나 이미지를 "선정적인", "폭력적인", "사회통념상 용납되지 않는 장면" 을 추가할 수도 있습니다.
영상을 나아가 콘텐츠내의 기능또한 수정이 가능하며 확률형 아이템등의 확률을 조작하는등 위험요소가 너무 많습니다.

**2.취약점 발견시 위험성**
 크래커는 현재 제페토 월드페키지에서 취약점을 발견하여, 유저의 정보(은행정보나 통신기록등)을 획득한다고해도 가져갈 수 없기때문에 이득을 취하기 매우 어렵습니다. 정보탈취로 금전적 이득을 취할 수 없는 플랫폼의 경우에는 크래커는 프로그램을 타겟으로 삼지 않습니다.
취약점이 발견된 시점에서 문제가 되기도 하지만 통신이 불가능하다면 피해를 최소화 할 수 있습니다.

**3.유니티 페키지로 만들어진 월드의 엄청난 자유도**
 제페토월드API는 엄청난 자유도를 가지고 있습니다. 통신은 엄청난 자유도에 위험을 더하는 장치로, 위의 위험요소 외에도 기상천외한 방식으로 유저의 정보를 무단 획득하거나, 콘텐츠의 건전한 성장을 방해하는 악의적인 콘텐츠를 추가 할 수 있습니다.

**추가로 단방향 통신의 경우입니다.**
 네트워크를 정확히 이해하고 완벽하게 단방향 통신으로 만드는게 아니라면 단순 로그수집만 한다고 해도 정보를 주고받는다는점은 변하지 않습니다. 그렇게 만든다고 해도 제페토에서는 이게 단방향 통신만 하는지 특정상황에 양방향 통신을 하는지 확실하게 검사하기 어려울것이라고 생각합니다.

1.위험요소가 많음
2.문제 발생시 피해 최소화 가능
3.위험요소를 예상할 수 없는 엄청난 자유도를가진 월드개발
+단방향 통신도 인터넷 프로토콜은 양방향 통신을 하고있을 가능성이 있음

완벽한 자유를 보장하고 서버가 책임을 지는 인터넷과 달리, 제페토는 인터넷 공급업체가 아닙니다.
서버가 무법지대가 되지 않도록 관리해야할 필요가 있으며, 자유를 이용한 개인의 크래킹으로 발생한 문제에 큰 타격을 입습니다.
개인적으로 외부통신은 지속적으로 허락되지 않기를 바라며, 콘텐츠에 필요한 통신이라면 월드 멀티플레이를 이용하는것으로 해결 가능하다고 생각합니다.

여러분들은 어떠한 제약사항에서도 좋은 콘텐츠를 만들어낼 수 있는 "개발자"라고 생각합니다. 한명의 유저로서 여러분들이 제작해주실 콘텐츠를 기대합니다. 잼 충전하고 기다리고있습니다!

틀린 정보가 있다면 댓글달아주세요 :D

**Comment by runasy-koonta** [comment]
외부 통신을 막는 것에 대한 의도에 대해서는 충분히 이해하지만, 최소한의 analytics 정보는 제공해주었으면 좋겠네요…
프로덕션에서 오류가 나도 파악이 안 되니까요. 최소한 error reporting 메뉴라도 있어서 stack trace까진 아니더라도 스크린샷 + 상황 기재란(유저입력)을 통해 크리에이터 콘솔에서 확인할 수 있었으면 좋겠습니다.

**Reply by x4cf** [reply]
오류 리포트 메일을 받을 방법이 있으면 좋을것같긴해요. 유저랑 소통할 방법이없으니까 유저들이 잘 플레이 중인지, 생각못한 오류를 찾지는 않았을지 궁금하기도 하구요 :D
리포팅 기능은 필요한 부분을 구체화 시켜서 깃에 올려보는게 좋을것같아요!

**Reply by cheolhui-lee** [reply]
안녕하세요.
의견 주신대로 예외 상황에 대한 정보 접근의 한계에 대해서는 저희도 불편함을 인지하고 있는 상황입니다.
그래서 제한적이나마 일부 로그나 오류 상황을 수집해서 크리에이터분들께 제공할 수 있는 방법을 지속적으로 고민하고 있습니다.
당장은 불편하시겠지만 준비가 될 때까지 양해 부탁드립니다.
고맙습니다.

**Reply by runasy-koonta** [reply]
@cheolhui-lee 피드백 감사드립니다. 에러 리포트 기능이 있으면 유저 피드백을 더욱 적극적으로 수렴하여 더욱 높은 퀄리티의 월드를 개발할 수 있을 것 같습니다. 빠른 시일 내에 추가되기를 기대합니다.
